src/components/LayoutDialog.vue

```diff
@@ -80,22 +83,71 @@ function handleImageUpload(event: Event) {
   if (input.files && input.files[0]) {
     const file = input.files[0];
     const reader = new FileReader();
-    
+
     reader.onload = (e) => {
       if (e.target?.result) {
-        layout.backgroundImage = {
-          image: e.target.result as string,
-          x: layout.backgroundImage?.x || 0,
-          y: layout.backgroundImage?.y || 0,
-          width: layout.backgroundImage?.width || 10,
-          height: layout.backgroundImage?.height || 10
+        const map_image = new Image();
+        map_image.onload = async () => {
+          // get real width and height
+          const imageWidth = map_image.naturalWidth;
+          const imageHeight = map_image.naturalHeight;
+          // populate layout background image
+          layout.backgroundImage = {
+            image: map_image.src,
+            x: layout.backgroundImage?.x || 0,
+            y: layout.backgroundImage?.y || 0,
+            width: layout.backgroundImage?.width || imageWidth,
+            height: layout.backgroundImage?.height || imageHeight,
+            natural_width: imageWidth,
+            natural_height: imageHeight
```

Comment:
if you want to add property to BackgroundImage please add it to interface BackgroundImage

Reply:
```
  public backgroundImage = ref<BackgroundImage>({
    image: "",
    x: 0,
    y: 0,
    width: 10,
    height: 10,
    naturalHeight: 0,
    naturalWidth: 0
  });
```